src/mds/FSMap.cc

```diff
@@ -130,8 +130,8 @@ void FSMap::dump(Formatter *f) const
   // Use 'default' naming to match 'set-default' CLI
   f->dump_int("default_fscid", legacy_client_fscid);
 
-  f->open_object_section("compat");
-  compat.dump(f);
+  f->open_object_section("default_compat");
```

Comment:
```
2021-04-02T19:55:48.100 INFO:tasks.cephfs_test_runner:test_full_health (tasks.mgr.dashboard.test_health.HealthTest) ... FAIL
2021-04-02T19:55:48.101 INFO:tasks.cephfs_test_runner:
2021-04-02T19:55:48.101 INFO:tasks.cephfs_test_runner:======================================================================
2021-04-02T19:55:48.102 INFO:tasks.cephfs_test_runner:FAIL: test_full_health (tasks.mgr.dashboard.test_health.HealthTest)
2021-04-02T19:55:48.102 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2021-04-02T19:55:48.102 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2021-04-02T19:55:48.103 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_3d0f85a6c3046372f708a5f2893231db573d2f75/qa/tasks/mgr/dashboard/helper.py", line 480, in assertSchema
2021-04-02T19:55:48.103 INFO:tasks.cephfs_test_runner:    return _validate_json(data, schema)
2021-04-02T19:55:48.103 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_3d0f85a6c3046372f708a5f2893231db573d2f75/qa/tasks/mgr/dashboard/helper.py", line 710, in _validate_json
2021-04-02T19:55:48.104 INFO:tasks.cephfs_test_runner:    for key, sub_schema in schema.sub_elems.items()
2021-04-02T19:55:48.104 INFO:tasks.cephfs_test_runner:tasks.mgr.dashboard.helper._ValError: In `input['fs_map']`: missing keys: {'compat'}
2021-04-02T19:55:48.104 INFO:tasks.cephfs_test_runner:
2021-04-02T19:55:48.104 INFO:tasks.cephfs_test_runner:During handling of the above exception, another exception occurred:
2021-04-02T19:55:48.105 INFO:tasks.cephfs_test_runner:
2021-04-02T19:55:48.105 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2021-04-02T19:55:48.105 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_3d0f85a6c3046372f708a5f2893231db573d2f75/qa/tasks/mgr/dashboard/test_health.py", line 273, in test_full_health
2021-04-02T19:55:48.106 INFO:tasks.cephfs_test_runner:    self.assertSchema(data, schema)
2021-04-02T19:55:48.106 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_3d0f85a6c3046372f708a5f2893231db573d2f75/qa/tasks/mgr/dashboard/helper.py", line 482, in assertSchema
2021-04-02T19:55:48.106 INFO:tasks.cephfs_test_runner:    self.assertEqual(data, str(e))
2021-04-02T19:55:48.107 INFO:tasks.cephfs_test_runner:AssertionError: {'health': {'status': 'HEALTH_OK', 'check[322522 chars]: 0}} != "In `input['fs_map']`: missing keys: {'compat'}"
2021-04-02T19:55:48.107 INFO:tasks.cephfs_test_runner:
```
see https://pulpito.ceph.com/kchai-2021-04-02_15:50:13-rados-wip-kefu-testing-2021-04-02-1506-distro-basic-gibba/6017625/

Reply:
Thanks @tchaikov . I reverted the name change for the JSON.